src/DataGeneration/bootstrap.jl

```diff
@@ -28,17 +28,17 @@ kwargs = Dict(:input_data=>input_data, :n=>20, :block_size=>4);
 input2 = BootstrapInput{MovingBlock}(;kwargs...)
 ```
 """
```

Comment:
E.g. here, for the documentation, I would write something like : 

```
"""
    BootstrapInput(input_data, bootstrap_method::<:TSBootMethod; kwargs...)
    BootstrapInput{T <: TSBootMethod}(; kwargs...)
Contains the parameters needed to perform block bootstrap of type T to be used by `makedata()` 
function. T can be any subtype of TSBootMethod: Stationary, MovingBlock, or CircularBlock.
## Keyword Arguments
- `input_data`: data to be resampled. Must be a 1-D array
- `bootstrap_method`: Type of time series bootstrap to use. Must be subtype of TSBootMethod.
- `n`: size of resampled output data. Default: 100
- `block_size`: block size to use. Defaults to the optimal block length using `opt_block_length()`
## Examples
```julia
	@@ -28,17 +28,17 @@ kwargs = Dict(:input_data=>input_data, :n=>20, :block_size=>4);
input2 = BootstrapInput{MovingBlock}(;kwargs...)
```
"""
```

There is no need to clutter your documentation with parametric types, that would not help readability. Your code now accepts anything for `block_size`, so no need to say `block_size::TF` where `TF` would be a float-like type.

Reply:
Great, this is what I had thought of too, so now all the docstrings should be accurate.